jms/org.wso2.carbon.transport.jms/src/test/java/org/wso2/carbon/transport/jms/test/util/JMSTestUtils.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.transport.jms.test.util;
+
+import org.wso2.carbon.transport.jms.utils.JMSConstants;
+
+import java.util.HashMap;
+
+/**
+ * This class handles the utils methods that are needed for jms transport testing.
+ */
+public class JMSTestUtils {
+
+    /**
+     * Creates a parameter map for listening to a particular jms destination.
+     *
+     * @param destinationName   Particular jms destination name.
+     * @param connectionFactory Connection Factory name.
+     * @param destinationType   Destination type, whether queue or topic
+     * @param jmsMode           jms acknowledgement mode.
+     * @return a hash map of the jms properties that is needed to listen to a particular queue or topic.
+     */
+    public static HashMap<String, String> createJMSListeningParameterMap(String destinationName, String
```

Comment:
Return type should be a Map instead of a HashMap as the best practice to use in an API method signature